core/src/commonMain/kotlin/com/revenuecat/purchases/kmp/Purchases.kt

```diff
@@ -411,6 +412,74 @@ public expect class Purchases {
         onSuccess: (storeTransaction: StoreTransaction) -> Unit,
     )
 
+    /**
+     * iOS only. Requires iOS 18.0+ and StoreKit 2 and emits an error if these requirements aren't
+     * met.
+     *
+     * @param product: The product to check for eligible win-back offers on.
+     * @param onError Will be called if an error occurs, providing a [PurchasesError] describing
+     * what went wrong.
+     * @param onSuccess Will be called if the function completes successfully, including details
+     * on the [WinBackOffer]s that the subscriber is eligible for on the provided product.
+     */
+    public fun getEligibleWinBackOffersForProduct(
+        storeProduct: StoreProduct,
+        onError: (error: PurchasesError) -> Unit,
+        onSuccess: (List<WinBackOffer>) -> Unit,
+    )
+
+    /**
+     * iOS only. Requires iOS 18.0+ and StoreKit 2 and emits an error if these requirements aren't
+     * met.
+     *
+     * @param packageToCheck: The package to check for eligible win-back offers on.
+     * @param onError Will be called if an error occurs, providing a [PurchasesError] describing
+     * what went wrong.
+     * @param onSuccess Will be called if the function completes successfully, including details
+     * on the [WinBackOffer]s that the subscriber is eligible for on the provided package.
+     */
+    public fun getEligibleWinBackOffersForPackage(
+        packageToCheck: Package,
```

Comment:
In the JS hybrids, this parameter is named `aPackage`, but I went with `packageToCheck` here since it felt more in line with the `packageToPurchase` parameter from the purchase package functions.